### PR TITLE
Feature data model changes #55

### PR DESCRIPTION
**Pull request by lkolacz**
##### Feature list
* Add docstrings to each model
* PartnerProject - change location to locations
* Indicator - add frequency of reporting to it
* active flag on disaggregation on blueprint
* move disagg and disagg options on reportable model to separate models than jsonfield.

##### Progress checker
- [X] Django

### CODE REVIEW

File: django_api/django_api/apps/indicator/models.py [hunk 1]
```diff
@@ -160,10 +180,37 @@ def status(self):
...
     indicator_report = models.ForeignKey(IndicatorReport, related_name="indicator_location_data")
     location = models.ForeignKey('core.Location', related_name="indicator_location_data")
 
     disaggregation = JSONField(default=dict)
 
     def __unicode__(self):
         return "{} Location Data for {}".format(self.location, self.indicator_report)
+
+
+class Disaggregation(TimeStampedModel):
```
Comment (n1207n) [inline django_api/django_api/apps/indicator/models.py]: This is a great start for our disaggregation features. ;)

One comment I would like to put is to have a safeguard on duplicate disaggregation with same name label.

We could have `Age` disaggregation for Reportable 1 and Reportable 2 as well. They both have same Disaggregation label `Age` but we need to differentiate them somehow. We can enforce name and its reportable id as unique-together constraint in Django DB model.

Let's put unique_together constraint on Disaggregation model.

File: django_api/django_api/apps/indicator/models.py [hunk 2]
```diff
@@ -160,10 +180,37 @@ def status(self):
...
+        return "Disaggregation <pk:%s>" % self.id
+
+
+class DisaggregationValue(TimeStampedModel):
+    """
+    Disaggregation Value module. For example: Gender <Male, Female, Other>
+    """
+    disaggregation = models.ForeignKey(Disaggregation, related_name="disaggregation_value")
+    value = models.CharField(max_length=255, null=True, blank=True)
+    active = models.BooleanField(default=False)
```
Comment (n1207n) [inline django_api/django_api/apps/indicator/models.py]: We'll have to put post_save Django signal whenever `Disaggregation` gets saved.

The signal needs to look at its active field and if it's False, then we need to update active field on all of `DisaggregationValue` models linked to that `Disaggregation.

File: django_api/django_api/apps/indicator/models.py (hunk 1 above)
Comment (lkolacz) [inline django_api/django_api/apps/indicator/models.py]: done

File: django_api/django_api/apps/indicator/models.py (hunk 2 above)
Comment (lkolacz) [inline django_api/django_api/apps/indicator/models.py]: done

**Comment by lkolacz** [comment]
@n1207n .. please make a check again :) .. thx!

File: django_api/django_api/apps/indicator/signals.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from __future__ import unicode_literals
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from .models import DisaggregationValue, Disaggregation
+
+
+@receiver(post_save, sender=Disaggregation, dispatch_uid="disable_disagg_value")
+def disable_disagg_value(sender, instance, **kwargs):
+    if instance.active is False:
```
Comment (n1207n) [inline django_api/django_api/apps/indicator/signals.py]: Use `==` instead. `is` keyword is technically checking instance identity, along with value.

Comment (lkolacz) [inline django_api/django_api/apps/indicator/signals.py]: done

**Comment by chachra** [comment]
@n1207n lets merge that in asap / today.

**Pull request merged**